cmd/csi-driver/diag/hpe-logcollector.sh

```diff
@@ -73,21 +73,44 @@ if [ -d $directory ]; then
     #HPE Logs
 	cp -r /var/log/hpe-*.log $directory > /dev/null 2>&1
 
+    #Nimble logs, including CSP
+	cp -r /var/log/nimble-*.log $directory > /dev/null 2>&1
+
+    #Collect CSI sidecar container logs, if kubectl is available
+    podlist=`kubectl get pods -l app=hpe-csi-controller -A -o jsonpath='{range .items[*]}{.metadata.name}{" "}{.metadata.namespace}{"\n"}{end}' 2>/dev/null`
+    if [[ $? -eq 0 ]]
+    then
+        while read -r podname podnamespace
+        do
+            containerlist=`kubectl get pod $podname -n $podnamespace -o jsonpath={.spec.containers[*].name} 2>/dev/null`
+            if [[ $? -eq 0 ]]
+            then
+                for containername in $containerlist
+                do
+                    # The hpe-csi-driver log is collected separately
+                    if [[ "$containername" != "hpe-csi-driver" ]]
+                    then
+                        timeout 30 kubectl logs $podname -n $podnamespace -c $containername > $directory/$podname.$podnamespace.$containername.log 2>&1
+                    fi
+                done
+            fi
+        done < <(printf '%s\n' "$podlist")
+    fi
```

Comment:
This section should be in the master script which has kubectl access. You can find it at https://raw.githubusercontent.com/hpe-storage/csi-driver/master/hpe-logcollector.sh 
and equivalent documentation at https://scod.hpedev.io/csi_driver/diagnostics.html#log_collector
The current script is packaged inside the container which would not have access to kubectl. 
Did this work for you?